lib/index.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const Bluebird = require('bluebird');
+
+exports.register = (server, options, next) => {
+
+  server.ext('onPreResponse', (request, reply) => {
+
+    const settings = request.route.settings.plugins.totalCount;
+    const credentials = request.auth.credentials;
+    const query = request.query;
+    const filter = request.query.filter || {};
+
+    if (settings && credentials && query.include && query.include.indexOf('total_count') !== -1 && request.response.source) {
```

Comment:
Optional: would it make sense to just use `Joi` validation for the configuration?